cirq-rigetti/requirements.txt

```
@@ -1 +1 @@
-pyquil>=3.2.0,<4.0.0
+pyquil>=4.10.0,<5.0.0
```

pyquil-4.1.0 can be installed for Python 3.12.1 (4.10.0 cannot per https://github.com/rigetti/pyquil/issues/1736) and passes unit tests.
The expression below matches 4.1.0 and any later releases that would declare compatibility with Python `3.12.*`.
It should thus unblock failing pytests with Python 3.12.

```suggestion
pyquil~=4.1
```

---

I've changed this to `>=4.11` as there's been a lot of bug fixes and quality-of-life improvements since 4.1, but this new version bound opens up the python 3.12 requirement.